sequence.go

```diff
@@ -26,7 +28,7 @@ func (s *Sequence) Value(state ReadOnlyState) uint64 {
 	return seq
 }
 
-func (s *Sequence) Next(state State) uint64 {
+func (s *Sequence) Next(state State, kvStore store.KVStore, isCheckTx bool) uint64 {
```

Comment:
`Sequence` should operate on `KVStoreReader` & `KVStoreWriter`, not `ReadOnlyState` & `State`.
Also it shouldn't have to care about CheckTx or DeliverTx, that should be the middleware's job.

Reply:
Done